relays/ethereum/src/headers.rs

```diff
@@ -356,6 +357,109 @@ impl<P: HeadersSyncPipeline> QueuedHeaders<P> {
 		}
 	}
 
+	/// When header completion data is sent to target node.
+	pub fn header_completed(&mut self, id: &HeaderId<P::Hash, P::Number>) {
+		if self.completion_data.remove(id).is_some() {
+			log::debug!(
+				target: "bridge",
+				"Sent completion data to {} for header: {:?}",
+				P::TARGET_NAME,
+				id,
+			);
+
+			// transaction can be dropped by target chain nodes => it would never be mined
+			//
+			// in current implementation the sync loop would wait for some time && if best
+			// **source** header won't change on **target** node, then the sync will be restarted
+			// => we'll resubmit the same completion data again (the same is true for submitted
+			// headers)
+			//
+			// the other option would be to track emitted transactions at least on target node,
+			// but it won't give us 100% guarantee anyway
+			//
+			// => we're just dropping completion data just after it has been submitted
+		}
+	}
+
+	/// When incomplete headers ids are receved from target node.
+	pub fn incomplete_headers_response(&mut self, ids: HashSet<HeaderId<P::Hash, P::Number>>) {
```

Comment:
Need a test for case when two relays are running and relay#2 completes some header, making some descendant incomplete => make sure that relay#1 will also mark it incomplete, without submitting children.